File: frontend/impacta-garagetrack/src/types/parkingTax.ts
export type ParkingTaxType = 'Hour' | 'AfterHour' | 'Daily'

export interface ParkingTax {
  id: number
  type: ParkingTaxType
  minutes: number | null
  value: number
  companyId: number
  createdAt: string
  updatedAt: string
}

export interface ParkingTaxRequest {
  type: ParkingTaxType
  minutes: number | null
  value: number
}

export const PARKING_TAX_TYPE_LABELS: Record<ParkingTaxType, string> = {
  Hour: 'Hora',
  AfterHour: 'Hora Adicional',
  Daily: 'Diária',
}

File: backend/Impacta.GarageTrack.System.Api/Impacta.GarageTrack.System.Api/Domain/ParkingTax/Entities/ParkingTax.cs
using Impacta.GarageTrack.System.Api.Domain.Kernel;

namespace Impacta.GarageTrack.System.Api.Domain.ParkingTax.Entities
{
    public class ParkingTax : EntityBase<long>
    {
        private ParkingTax()
        {
            Type = ParkingTaxType.Hour;
            Value = 0;
            CompanyId = 0;
        }

        public ParkingTax(ParkingTaxType type, int? minutes, int? fromHours, decimal value, long companyId)
        {
            Type = type;
            Minutes = minutes;
            FromHours = fromHours;
            Value = value;
            CompanyId = companyId;
        }

        public ParkingTaxType Type { get; set; }
        public int? Minutes { get; set; }
        public int? FromHours { get; set; }
        public decimal Value { get; set; }
        public long CompanyId { get; set; }
    }

    public enum ParkingTaxType
    {
        Hour = 1,
        AfterHour = 2,
        Daily = 3
    }
}
